### PR TITLE
MGDAPI-4618 - address all linter issues with golangci-lint

### PR DESCRIPTION
**Pull request by kramaranya**
# Issue link

[Linter Epic](https://issues.redhat.com/browse/MGDAPI-4593)

# What

Address all the errors from the Linter Epic: staticcheck and gosimple

# Verification steps

- Checkout this PR
- Run `make test/lint`
- Confirm there are no errors in the output
- Code review

### CODE REVIEW

**Review by Boomatang — CHANGES_REQUESTED** [review]
Couple of small points.

File: pkg/products/threescale/reconciler_test.go [hunk 1]
```diff
@@ -3041,12 +3041,12 @@ func TestReconciler_ping3scalePortals(t *testing.T) {
...
 				serverClient: func() k8sclient.Client {
 					mockClient := moqclient.NewSigsClientMoqWithScheme(scheme, systemSeed)
-					mockClient.ListFunc = func(ctx context.Context, list k8sclient.ObjectList, opts ...k8sclient.ListOption) error {
+					mockClient.ListFunc = func(ctx context.Context, list k8sclient.ObjectList, opts ...k8sclient.ListOption) error { //nolint:staticcheck
 						listOpts := k8sclient.ListOptions{}
 						listOpts.ApplyOptions(opts)
 
 						if listOpts.LabelSelector.Matches(labels.Set(map[string]string{"zync.3scale.net/route-to": labelRouteToSystemMaster})) {
-							list = &routev1.RouteList{ // nolint:ineffassign
+							list = &routev1.RouteList{ // nolint
```
[the remark this answers is not in the document]
Comment (kramaranya) [reply pkg/products/threescale/reconciler_test.go]: Updated

**Comment by Fiona-Waters** [comment]
Followed verification steps and can confirm that there were no errors in the output. Code changes look good. Will leave it to  @Boomatang to lgtm if he is happy with newest changes.

**Review by Boomatang — APPROVED** [review]
Changes look good to me
/lgtm

**Pull request merged**